static/js/context.js

```diff
@@ -1,14 +1,320 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from "react";
+
 const ContentLanguageContext = React.createContext({
   language: "english",
 });
-ContentLanguageContext.displayName = 'ContentLanguageContext'; //This lets us see this name in the devtools
+ContentLanguageContext.displayName = "ContentLanguageContext"; //This lets us see this name in the devtools
 
-const AdContext = React.createContext({
-});
-AdContext.displayName = 'AdContext';
+const AdContext = React.createContext({});
+AdContext.displayName = "AdContext";
+
+const StrapiDataContext = React.createContext({});
+StrapiDataContext.displayName = "StrapiDataContext";
+
+const transformValues = (obj, callback) => {
+  const newObj = {};
+  for (let key in obj) {
+    newObj[key] = obj[key] !== null ? callback(obj[key]) : null;
+  }
+  return newObj;
+};
+
+export function replaceNewLinesWithLinebreaks(content) {
```

Comment:
I'm using it as a utility function for processing the markdown text that's returned from Strapi. In markdown, to create a line break, you need to add 2 spaces at the end of a line and that's how the current react-markdown plugin used in InterfaceText handles that. Since Strapi is intended to be used by other people at Sefaria besides engineers, that is really counterintuitive and they will likely not know or remember to add spaces.